hack/deploy-cluster.sh

```diff
@@ -17,30 +17,46 @@ set -e
 REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
 RESOURCE_GROUP_NAME=${RESOURCE_GROUP_NAME:-""}
 LOCATION=${LOCATION:-""}
+
+
+#check for variables is initialized or not
+
+
 SUBSCRIPTION_ID=${SUBSCRIPTION_ID:-""}
 CLIENT_ID=${CLIENT_ID:-""}
 CLIENT_SECRET=${CLIENT_SECRET:-""}
 TENANT_ID=${TENANT_ID:-""}
 USE_CSI_DEFAULT_STORAGECLASS=${USE_CSI_DEFAULT_STORAGECLASS:-""}
+ENABLE_AVAILABILITY_ZONE=${ENABLE_AVAILABILITY_ZONE:-""}
 
-# Check for variables is initialized or not
-if [ -z "$LOCATION" ] || [ -z "${SUBSCRIPTION_ID}" ] || [ -z "${CLIENT_ID}" ] || [ -z "${CLIENT_SECRET}" ] || [ -z "${TENANT_ID}" ] || [ -z "${USE_CSI_DEFAULT_STORAGECLASS}" ]; then
-  echo "SUBSCRIPTION_ID, CLIENT_ID, TENANT_ID, CLIENT_SECRET ,LOCATION and USE_CSI_DEFAULT_STORAGECLASS must be specified"
+# check for variables is initialized or not
+if [ -z "$LOCATION" ] || [ -z "${SUBSCRIPTION_ID}" ] || [ -z "${CLIENT_ID}" ] || [ -z "${CLIENT_SECRET}" ] || [ -z "${TENANT_ID}" ] || [ -z "${USE_CSI_DEFAULT_STORAGECLASS}" ] || [ -z "${ENABLE_AVAILABILITY_ZONE}"]; then
+  echo "SUBSCRIPTION_ID, CLIENT_ID, TENANT_ID, CLIENT_SECRET ,LOCATION ,USE_CSI_DEFAULT_STORAGECLASS and ENABLE_AVAILABILITY_ZONE must be specified"
```

Comment:
let's make ENABLE_AVAILABILITY_ZONE  optional. When ENABLE_AVAILABILITY_ZONE  is not set, disable the AZ

Reply:
okay , to set that the user can put ENABLE_AVAILABILITY_ZONE=yes